src/lib/link.ts

```diff
@@ -44,10 +44,50 @@ export class Link {
     connA: string,
     connB: string
   ): Promise<Link> {
-    // so they are no marked unused variables
-    [nodeA, nodeB, connA, connB];
-
-    throw new Error('not yet implemented');
+    const [clientIdA, clientIdB] = await createClients(nodeA, nodeB);
+    const [
+      { connection: connectionA },
+      { connection: connectionB },
+    ] = await Promise.all([
+      nodeA.query.ibc.connection.connection(connA),
+      nodeB.query.ibc.connection.connection(connB),
+    ]);
+    if (!connectionA) {
+      throw new Error(`Connection not found for ID ${connA}`);
+    }
+    if (!connectionB) {
+      throw new Error(`Connection not found for ID ${connB}`);
+    }
+    if (!connectionA.counterparty) {
+      throw new Error(`Counterparty not found for connection with ID ${connA}`);
+    }
+    if (!connectionB.counterparty) {
+      throw new Error(`Counterparty not found for connection with ID ${connB}`);
+    }
+    if (connectionA.clientId !== connectionB.counterparty.clientId) {
+      throw new Error(
+        `Client ID ${connectionA.clientId} for connection with ID ${connA} does not match counterparty client ID ${connectionB.counterparty.clientId} for connection with ID ${connB}`
+      );
+    }
+    if (connectionB.clientId !== connectionA.counterparty.clientId) {
+      throw new Error(
+        `Client ID ${connectionB.clientId} for connection with ID ${connB} does not match counterparty client ID ${connectionA.counterparty.clientId} for connection with ID ${connA}`
+      );
+    }
+    const [clientStateA, clientStateB] = await Promise.all([
+      nodeA.query.ibc.client.stateTm(connectionA.clientId),
+      nodeB.query.ibc.client.stateTm(connectionB.clientId),
+    ]);
+    // TODO: What should be done with these?
```

Comment:
Minimally, check the remote chain id. Secondly, query the latest consensus state, and you can look for a matching header on the remote node to be thorough